packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -0,0 +1,149 @@
+/**
+ * @fileoverview Rule to flag statements that use magic numbers (adapted from https://github.com/danielstjules/buddy.js)
+ * @author Scott O'Hara
+ */
+
+import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import baseRule from 'eslint/lib/rules/no-magic-numbers';
+import * as util from '../util';
+import { JSONSchema4 } from 'json-schema';
+
+type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+// Original schema properties from the base rule
+const { properties } = (baseRule.meta.schema as JSONSchema4[])[0];
+
+// Extend base schema with additional property to ignore TS numeric literal types
+if (properties) {
+  properties.ignoreNumericLiteralTypes = {
```

Comment:
Unless I'm mistaken - node will reuse this object for other requires of the same file.
I would suggest not modifying the base object so that we don't inadvertently affect the operation of anything else.

Reply:
Good point. Updated to spread the base object + new options onto a new object, so as to not modify the original.